src/components/organisms/MapBoxMap/MapboxMap.tsx

```diff
@@ -44,57 +55,175 @@ const MapboxMap: React.FC = () => {
-                        addPredictionLayer(map, jobs[0], regionId)
-                    })
-                })
-            })
-            .catch((error) => console.error('Failed to load regions on map:', error))
-
         return () => map.remove()
     }, [])
 
+    const { isPending: fetchingAoiCenters, data } = useQuery({
+        queryKey: ['regions'],
+        // queryFn: fetchRegions,
+        queryFn: async () => await fetchRegions(),
+    })
+
+    useEffect(() => {
+        if (data && map) {
+            loadRegionsOnMap(data)
+        }
+    }, [data, map])
+
+    //fails on hot reload, page reload works fine
+    function loadRegionsOnMap(regions: any) {
+        if (!map) {
+            console.error('Map is not initialized')
+            return
+        }
+        addRegionLayer(map, regions)
+
+        map.on('click', 'regions', (e) => {
+            const regionId = e.features![0].properties!.id
+            const regionName = e.features![0].properties!.name
+            const regionSize = e.features![0].properties!.area_km2
+            const regionPolygon: Polygon = JSON.parse(e.features![0].properties!.polygon)
+
+            const regionData: IRegionData = {
+                id: regionId,
+                name: regionName,
+                areaSize: regionSize,
+                polygon: regionPolygon,
+            }
+
+            //this causes the react query hook to refetch the data with the new regionId
+            setRegionId(e.features![0].properties!.id)
+            setRegionData(regionData)
+
+            return
+        })
+    }
+
+    async function fetchRegionDatetimesWithTimeout(regionId: number) {
+        if (!map) return []
+        if (!regionId) return []
+
+        //timout to simulate loading
+        //await new Promise((resolve) => setTimeout(resolve, 1000))
+
+        const regionDatetimes = await fetchRegionDatetimes(regionId)
+        const regionJobs = regionDatetimes.map((regionDatetimes) => regionDatetimes.timestamp)
+
+        setFetchingPredictionsForDay(regionJobs[0])
+        //addPredictionLayer(map, regionJobs[0], regionId)
+
+        map.setLayoutProperty('regions', 'visibility', 'none')
+        map.fitBounds(getBoundingBox(regionData?.polygon!))
+
+        addPolygonLayer(map, regionData?.polygon!)
+        openSidebar()
+
+        if (regionData && regionJobs) {
+            const newRegionProps: IRegionProperties = {
+                id: regionData.id,
+                name: regionData.name,
+                areaSize: regionData.areaSize,
+                jobs: regionJobs,
+            }
+            setRegionProps(newRegionProps)
+        }
+
+        return regionDatetimes
+    }
+
+    const { isPending: fetchingRegionData } = useQuery({
+        queryKey: [`regionData${regionId}`, regionId],
+        queryFn: async () => await fetchRegionDatetimesWithTimeout(regionId!),
+        enabled: regionId !== null,
+        refetchOnWindowFocus: false,
+    })
```

Comment:
as explained in an earlier comment put the query inside a function and call it directly not through setting some state variable, rather pass any needed parameters directly to the function

in addition rename isPending as it is not very obvious to understand what exactly is pending when you use it later on in the code